### PR TITLE
AYW Surveys: Enable all pre and post links

### PR DESCRIPTION
**Pull request by molly-moen**
Enable pre and post workshop links for all AYW (academic year workshop) surveys. Pre-surveys and kickoff post survey have been fully proofread and are ready to go. Post surveys are still in progress but the first workshop is not until 8/29 so it is safe to enable them now.

 ### Future work 
- update survey results view for academic year workshops to show foorm view
- deprecate old AYW links and clean up the workshop_daily_survey_controller (will be done in separate PR)
 
## Links

- [spec - link descriptions](https://docs.google.com/document/d/1uYKpTUS3eJOna9008p_6-30fEeX_vdPc0dEDEDc-MXU/edit)
- [jira](https://codedotorg.atlassian.net/browse/PLC-954)

## Testing story
Added unit tests for the new links and tested locally that an enrolled teacher can access the links as expected.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb [hunk 1]
```diff
@@ -1029,6 +1029,77 @@ class WorkshopDailySurveyControllerTest < ActionDispatch::IntegrationTest
       assert_select 'h1', text: 'Thank you for submitting today’s survey.'
     end
 
+    test 'AYW1 pre-survey link shows foorm survey' do
```
Comment (bencodeorg) [inline dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb:1032]: This is a crazy test file!

File: dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb [hunk 2]
```diff
@@ -1029,6 +1029,77 @@ class WorkshopDailySurveyControllerTest < ActionDispatch::IntegrationTest
...
+
+    test 'AYW1 post-survey link shows foorm survey for attended teacher' do
+      setup_academic_year_workshop
+      sign_in @enrolled_academic_year_teacher
+      create :pd_attendance,
+        session: @academic_year_workshop.sessions[0],
+        teacher: @enrolled_academic_year_teacher,
+        enrollment: @academic_year_enrollment
+
+      pre_survey_links = %w(/pd/AYW1/post/in_person /pd/AYW1/post/module/1 /pd/AYW1/post/module/2 /pd/AYW1/pre/module/1_2)
```
Comment (bencodeorg) [inline dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb]: Typo variable name? Also the last case is a pre-survey?

File: dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb [hunk 3]
```diff
@@ -1029,6 +1029,77 @@ class WorkshopDailySurveyControllerTest < ActionDispatch::IntegrationTest
...
+      create :pd_attendance,
+        session: @two_day_academic_year_workshop.sessions[0],
+        teacher: @enrolled_two_day_academic_year_teacher,
+        enrollment: @two_day_academic_year_enrollment
+      get '/pd/AYW1_2/post/in_person'
+      assert_response :success
+      assert_template :new_general_foorm
+    end
+
+    test 'invalid AYW survey links show appropriate errors' do
```
Comment (bencodeorg) [inline dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb]: Test name seems like a misnomer. Maybe `'workshop enrollee can only load appropriate survey links` that both checks they can't load survey(s) they shouldn't be able to load, and can load the survey they should be able to load?

Comment (bencodeorg) [reply dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb]: Or could make the test actually check when invalid AYW survey links are used (eg, `/AYW/this_is_an_invalid_link`

File: dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb (hunk 2 above)
Comment (molly-moen) [reply dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb]: ah yeah going too fast on this one :)

File: dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb (hunk 3 above)
Comment (molly-moen) [reply dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb]: yeah I will update, I originally planned to include more links and didn't update the title

**Review by bencodeorg — APPROVED** [review]

**Pull request merged**